src/infrastructure/content/NetworkMessageHandler.java

```diff
@@ -0,0 +1,247 @@
+package infrastructure.content;
+
+import networking.INotificationHandler;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.util.HashMap;
+import infrastructure.validation.logger.ILogger;
+import infrastructure.validation.logger.LogLevel;
+import infrastructure.validation.logger.LoggerFactory;
+import infrastructure.validation.logger.ModuleID;
+
+/**
+ * NetworkMessageHandler class implements the INotificationHandler interface.
+ * this includes one method i.e onMessageReceived()
+ * this class is used to notify and to send data to the UI module 
+ * as new user joined or as a user exit or as message sent by a user
+ * 
+ * @author Talha Yaseen
+ */
+public class NetworkMessageHandler implements INotificationHandler {
+		
+	/**
+	* Creating instance of class which implements ILogger interface
+	*/
+	private static ILogger logger = LoggerFactory.getLoggerInstance();
+	
+	/**
+	 * Accessing handlerMap hashmap from ContentCommunicator class to use it for sending message to UI
+	 */
+	private static HashMap<String, IContentNotificationHandler> handlerMap = ContentCommunicator.getHandlerMap();
+	
+	/**
+	 * Accessing imageMap hashmap from ContentCommunicator class to access image corresponding to username
+	 */
+	private static HashMap<String, String> imageMap = ContentCommunicator.getImageMap();
+	
+	/**
+	 * This variable will store the meta field of json string sent by the Networking module
+	 */
+	private static String metafield;
+	
+	/**
+	 * This array will store the jsonObject whose fields would be username and image
+	 */
+	private static JSONArray jsonObjectArray;
+	 
+	/**
+	 * This json object will store the elements of jsonObjectArray array
+	 */
+	private static JSONObject object;
+	
+	/**
+	 * This variable will store the username of client
+	 */
+	private static  String username;
+	
+	/**
+	 * This variable will store the image accessed by imageMap corresponding to username 
+	 */
+	private static String userimage;
+	
+	/**
+	 * Creating a HashMap to store userName and userImage of all active clients of a particular board.
+	 * Both userName and userImage are of String type
+	 */
+	private static HashMap<String, String> imagemap;
+	
+	/**
+	* This variable will store a string which will contains log message that to be passed to log method of logger
+	*/
+	private static String logMessage;
+	
+	/**
+	 * Getting handler of the UI module to send the data
+	 */
+	private static IContentNotificationHandler handler;
+	 
+	/**
+	 * @param String message - JSON string message (Meta fields are newUser or message or userExit)
```

Comment:
`String` should be removed. It won't be parsed correctly.
`@param message` is the correct way.

Reply:
resolved